sequencer/batchbuilder.go

```diff
@@ -171,11 +216,17 @@ func (s *Sequencer) isSequenceProfitable(ctx context.Context) bool {
 }
 
 func (s *Sequencer) processTxs(ctx context.Context, pendingTxs []*pool.Transaction) (
-	[]*state.ProcessTransactionResponse, []common.Hash, map[string]*state.ProcessTransactionResponse, error) {
+	processTxResponse, error) {
+	response := processTxResponse{
+		processedTxs:       nil,
+		processedTxsHashes: nil,
+		unprocessedTxs:     nil,
+		isUnprocessedBatch: true,
```

Comment:
if you change `isUnprocessedBatch -> isBatchProcessed`, you will not need to init this struct and could just return `&processTxResponse`